src/main/java/org/mitre/synthea/helpers/RandomCodeGenerator.java

```diff
@@ -1,191 +1,111 @@
-          + offset);
-      try {
-        response = responseCache.get(new ExpandInput(valueSetUri, offset));
-      } catch (ExecutionException e) {
-        throw new RuntimeException("Error expanding ValueSet", e);
-      }
-      validateExpansion(response.getExpansion());
-      total = response.getExpansion().getTotal();
-      count = response.getExpansion().getContains().size();
-      result.getContains().addAll(response.getExpansion().getContains());
-    } while ((offset + count) < total);
-    result.setTotal(total);
-
-    return result;
-  }
-
-  private static void validateExpansion(@Nonnull ValueSetExpansionComponent expansion) {
-    if (expansion.getContains().isEmpty()) {
-      throw new RuntimeException("ValueSet expansion does not contain any codes");
-    } else if (expansion.getTotalElement().isEmpty()) {
-      throw new RuntimeException("No total element in ValueSet expand result");
-    }
-  }
-
-  private static void validateContains(ValueSetExpansionContainsComponent contains) {
-    if (contains.getSystem() == null || contains.getCode() == null
-        || contains.getDisplay() == null) {
-      throw new RuntimeException(
-          "ValueSet contains element does not contain system, code and display");
-    }
-  }
-
-  private static class ExpandInput {
-
-    @Nonnull
-    private final String valueSetUri;
-    private final int offset;
-
-    public ExpandInput(@Nonnull String valueSetUri, int offset) {
-      this.valueSetUri = valueSetUri;
-      this.offset = offset;
-    }
-
-    @Nonnull
-    public String getValueSetUri() {
-      return valueSetUri;
-    }
-
-    public int getOffset() {
-      return offset;
-    }
-
-    @Override
-    public boolean equals(Object o) {
-      if (this == o) {
-        return true;
-      }
-      if (o == null || getClass() != o.getClass()) {
-        return false;
-      }
-      ExpandInput that = (ExpandInput) o;
-      return offset == that.offset
-          && valueSetUri.equals(that.valueSetUri);
-    }
-
-    @Override
-    public int hashCode() {
-      return Objects.hash(valueSetUri, offset);
-    }
-
-  }
+	public static String expandBaseUrl = Config.get("generate.terminology_service_url") + "/ValueSet/$expand?url=";
+
+	private static final Logger logger = LoggerFactory.getLogger(RandomCodeGenerator.class);
+	public static Map<String, List<Object>> codeListCache = new HashMap<>();
+	public static List<Code> selectedCodes = new ArrayList<>();
+
+	public static RestTemplate restTemplate = new RestTemplate();
+
+	/**
+	 * Gets a random code from the expansion of a ValueSet.
+	 *
+	 * @param valueSetUri
+	 *            the URI of the ValueSet
+	 * @param seed
+	 *            a random seed to ensure reproducibility of this result
+	 * @return the randomly selected Code
+	 */
+	@SuppressWarnings({ "unchecked", "static-access" })
+	public static Code getCode(String valueSetUri, long seed) {
+		expandValueSet(valueSetUri);
```

Comment:
Should there be some parameter validation here first?

Reply:
@hankwallace, now validating vaueSetUri, if it is wrong one using default code